### PR TITLE
Explicit mgmt cluster registration steps

### PR DESCRIPTION
**Pull request by ashleywang1**
BOT NOTES: 
resolves https://github.com/solo-io/gloo-mesh/issues/1384
resolves https://github.com/solo-io/gloo-mesh/issues/1428

### CODE REVIEW

**Comment by ashleywang1** [comment]
This is what it looks like:

Screen Shot 2021-04-13 at 4 00 21 PM

File: docs/content/setup/installation/enterprise_installation.md [hunk 1]
```diff
@@ -106,16 +106,44 @@ Note that the `gloo-mesh-enterprise` Helm chart bundles multiple components, inc
...
 
-```shell
+Case 1: Your Gloo Mesh control plane will be running on its own cluster.
+
+Case 2: You are running Gloo Mesh Enterprise's management plane on a cluster
+that you intend to register (i.e. also run a service mesh). In this case,
+set the `enterprise-networking.cluster` value to the cluster name you
+intend to set for the management cluster at registration time.
+
+Case 3: You intend to provide your own certificates. Note that
```
Comment (joekelley) [inline docs/content/setup/installation/enterprise_installation.md:119]: We should describe / link to steps users can take to configure gloo mesh to leverage user-provided certs, e.g. what names should users use for their secrets and what should those secrets contain

File: docs/content/setup/installation/enterprise_installation.md [hunk 2]
```diff
@@ -106,16 +106,44 @@ Note that the `gloo-mesh-enterprise` Helm chart bundles multiple components, inc
 
 4. Install
 
-{{% notice note %}}If you are running Gloo Mesh Enterprise's management plane on a cluster you intend to register (i.e. also run a service mesh), set the `enterprise-networking.cluster` value to the cluster name you intend to set for the management cluster at registration time. {{% /notice %}}
+There are several important helm values to modify, depending on your setup.
```
Comment (joekelley) [inline docs/content/setup/installation/enterprise_installation.md:109]: Let's link to our helm reference docs here: https://docs.solo.io/gloo-mesh/latest/reference/helm/gloo_mesh_enterprise/

And highlight that the user May have to modify some values; let's imply that our default settings are sufficient for typical deployments.

File: docs/content/setup/installation/enterprise_installation.md [hunk 3]
```diff
@@ -106,16 +106,45 @@ Note that the `gloo-mesh-enterprise` Helm chart bundles multiple components, inc
...
+There are several [helm values]({{< versioned_link_path fromRoot="/reference/helm/gloo_mesh_enterprise" >}}) that you may have
+to modify, depending on your setup. Here are several examples:
 
-```shell
+Case 1: Your Gloo Mesh control plane will be running on its own cluster.
+
+Case 2: You are running Gloo Mesh Enterprise's management plane on a cluster
+that you intend to register (i.e. also run a service mesh). In this case,
+set the `enterprise-networking.cluster` value to the cluster name you
+intend to set for the management cluster at registration time.
```
Comment (joekelley) [inline docs/content/setup/installation/enterprise_installation.md:117]: Let's be explicit that the user will still have to register the cluster even after they set the cluster name here. We got feedback from a user that they didn't know they still had to install the enterprise agent / add a kubernetes cluster resource representing the mgmt cluster.

Something like `In this case,
set the `enterprise-networking.cluster` value to the cluster name you
intend to set for the management cluster at registration time. Note that you still need to explicitly register the management cluster vie meshctl or helm.`

File: docs/content/setup/installation/enterprise_installation.md [hunk 4]
```diff
@@ -106,16 +106,45 @@ Note that the `gloo-mesh-enterprise` Helm chart bundles multiple components, inc
...
-{{% notice note %}}If you are running Gloo Mesh Enterprise's management plane on a cluster you intend to register (i.e. also run a service mesh), set the `enterprise-networking.cluster` value to the cluster name you intend to set for the management cluster at registration time. {{% /notice %}}
+There are several [helm values]({{< versioned_link_path fromRoot="/reference/helm/gloo_mesh_enterprise" >}}) that you may have
+to modify, depending on your setup. Here are several examples:
 
-```shell
+Case 1: Your Gloo Mesh control plane will be running on its own cluster.
+
+Case 2: You are running Gloo Mesh Enterprise's management plane on a cluster
+that you intend to register (i.e. also run a service mesh). In this case, set
+the enterprise-networking.cluster value to your management cluster name. Then, register
```
Comment (joekelley) [inline docs/content/setup/installation/enterprise_installation.md:116]:
```suggestion
the `enterprise-networking.cluster` value to your management cluster name. Then, register
```

**Comment by joekelley** [comment]
These changes look good, but could we also describe how users can go about providing helm values overrides in the `meshctl install` steps above? These options are available to meshctl users as well.

File: docs/content/setup/installation/enterprise_installation.md [hunk 5]
```diff
@@ -51,28 +51,54 @@ You can also quickly install like this:
...
-We are not going to change any of the default values in the underlying chart, so the only argument needed is the license key.
+Several ways to install using `meshctl` are:
 
 ```shell
-meshctl install enterprise --license $GLOO_MESH_LICENSE_KEY
+# Install to the currently selected Kubernetes context
+meshctl install enterprise --license=<my_license>
+
+# Install to and register the currently selected Kubernetes context
+meshctl install enterprise --license=<my_license> --register
```
Comment (joekelley) [inline docs/content/setup/installation/enterprise_installation.md:73]: I believe you need a cluster name in this case

File: docs/content/setup/installation/enterprise_installation.md [hunk 6]
```diff
@@ -51,28 +51,54 @@ You can also quickly install like this:
...
 ```shell
 GLOO_MESH_LICENSE_KEY=<your_key_here> # You'll need to supply your own key
 ```
 
-We are not going to change any of the default values in the underlying chart, so the only argument needed is the license key.
+Several ways to install using `meshctl` are:
 
 ```shell
-meshctl install enterprise --license $GLOO_MESH_LICENSE_KEY
+# Install to the currently selected Kubernetes context
```
Comment (joekelley) [inline docs/content/setup/installation/enterprise_installation.md:69]: could we use the tabbed view here instead of three inline commands?

**Comment by joekelley** [comment]
moved changes to https://github.com/solo-io/gloo-mesh/pull/1481

**Pull request closed without merging**